tests/test_edge_cases.py

```diff
@@ -926,11 +926,29 @@ def test_inheritance():
     class Foo(BaseModel):
         a: float = ...
 
-    class Bar(Foo):
+    with pytest.raises(
+        TypeError,
+        match=(
+            "Field 'a' defined on a base class was overridden by a non-annotated attribute. "
+            'All field definitions, including overrides, require a type annotation.'
+        ),
+    ):
+
+        class Bar(Foo):
+            x: float = 12.3
+            a = 123.0
+
+    class Bar2(Foo):
+        x: float = 12.3
+        a: float = 123.0
+
+    assert Bar2().model_dump() == {'x': 12.3, 'a': 123.0}
+
+    class Bar3(Foo):
         x: float = 12.3
-        a = 123.0
+        a: float = Field(default=123.0)
 
-    assert Bar().model_dump() == {'x': 12.3, 'a': 123.0}
+    assert Bar3().model_dump() == {'x': 12.3, 'a': 123.0}
```

Comment:
Can you make those numbers different for every Bar? I just want to make sure there’s no way we’re accidentally modifying the base class and getting the values from there